index.bs

```diff
@@ -308,18 +311,31 @@ Report long tasks {#report-long-tasks}
             1. Set |attribution|'s {{PerformanceEntry/startTime}} and {{PerformanceEntry/duration}} to 0.
-            1. If |culpritSettings| is not <code>null</code>, and |culpritSettings|'s <a>responsible browsing context</a> has a <a>browsing context container</a> that is an <{iframe}> element, then let |iframe| be that element, and perform the following steps:
-                1. Set |attribution|'s {{containerType}} attribute to "<code>iframe</code>".
-                1. Set |attribution|'s {{containerName}} attribute to the value of  |iframe|'s <{iframe/name}> content attribute, or <code>""</code> if the attribute is absent.
-                1. Set |attribution|'s {{containerSrc}} attribute to the value of |iframe|'s <{iframe/src}> content attribute, or <code>""</code> if the attribute is absent.
+            1. Set |attribution|'s {{containerType}} attribute to <code>"window"</code>.
+            1. Set |attribution|'s {{containerName}} and {{containerSrc}} attributes to the empty string.
+            1. If |culpritSettings| is not <code>null</code>:
+                1. Let |container| be |culpritSettings|'s <a>responsible browsing context</a>'s' <a>browsing context container</a>.
+                1. Assert: |container| is not <code>null</code>.
+                1. Set |attribution|'s {{containerId}} attribute to the value of |container|'s [=Element/ID=], or the empty string if the ID is unset.
+                1. If |container| is an <{iframe}> element:
+                    1. Set |attribution|'s {{containerType}} attribute to "<code>iframe</code>".
+                    1. Set |attribution|'s {{containerName}} attribute to the value of |container|'s <{iframe/name}> content attribute, or the empty string if the attribute is absent.
+                    1. Set |attribution|'s {{containerSrc}} attribute to the value of |container|'s <{iframe/src}> content attribute, or the empty string if the attribute is absent.
 
                     NOTE: it is intentional that we record the frame's <{iframe/src}> attribute here, and not its current URL, as this is meant primarily to help identify frames, and allowing discovery of the current URL of a cross-origin iframe is a security problem.
 
-                1. Set |attribution|'s {{containerId}} attribute to the value of |iframe|'s [=Element/ID=], or <code>""</code> if the ID is unset.
-            1. Otherwise, perform the following steps:
-                1. Set |attribution|'s {{containerType}} attribute to <code>"window"</code>.
-                1. Set |attribution|'s {{containerName}} attribute to <code>""</code>.
-                1. Set |attribution|'s {{containerSrc}} attribute to <code>""</code>.
+                1. If |container| is a <{frame}> element:
+                    1. Set |attribution|'s {{containerType}} attribute to "<code>frame</code>".
+                    1. Set |attribution|'s {{containerName}} attribute to the value of |container|'s {{Frame/name}} attribute.
```

Comment:
Is this correct? We use the IDL attribute instead of the content attribute, and that way we can skip the empty string part. Perhaps we can use this for the other object types as well?

Reply:
Not sure.

Reply:
No, this is not correct, since the public API could be overridden (e.g. `Object.defineProperty(frame, "name", { value: "anything" })`). It's best to stick to the content attributes as you did above.

Reply:
Done